lib/layer.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:ui';
+
+class Layer {
+  List<LayerProcessor> preProcessors = [];
+  List<LayerProcessor> postProcessors = [];
+
+  Picture _picture;
+
+  PictureRecorder _recorder;
+  Canvas _canvas;
+
+  void render(Canvas canvas, {double x = 0.0, double y = 0.0}) {
+    if (_picture == null) {
+      return;
+    }
+
+    canvas.save();
+    canvas.translate(x, y);
+
+    preProcessors.forEach((p) => p.process(_picture, canvas));
+    canvas.drawPicture(_picture);
+    postProcessors.forEach((p) => p.process(_picture, canvas));
+    canvas.restore();
+  }
+
+  Canvas get canvas {
+    assert(_canvas != null,
+        'Layer is not ready for rendering, call beginRendering first');
+    return _canvas;
+  }
+
+  void beginRendering() {
+    _recorder = PictureRecorder();
+    _canvas = Canvas(_recorder);
+  }
+
+  void finishRendering() {
+    _picture = _recorder.endRecording();
+
+    _recorder = null;
+    _canvas = null;
+  }
+}
+
+abstract class LayerProcessor {
+  void process(Picture pic, Canvas canvas);
+}
+
+class ShadowProcessor extends LayerProcessor {
```

Comment:
Maybe have the processors in separate files (could be done later when there are more of them).

Reply:
Sure, I could use some ideas to include some more processors :)